tools-local/tasks/mobile.tasks/CreateWasmBundle/DownloadTimeZoneData.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Net;
+using System.Reflection;
+using System.Diagnostics;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+public class DownloadTimeZoneData : Task
+{
+    [Required]
+    public string? InputDirectory { get; set; }
+
+    [Required]
+    public string? OutputDirectory { get; set; }
+
+    [Required]
+    public string? Version { get; set; }
+
+    private void DownloadTimeZoneDataSource() 
+    {
+        List<string> files = new List<string>() {"africa", "antarctica", "asia", "australasia", "etcetera", "europe", "northamerica", "southamerica", "zone1970.tab"};
+        using (var client = new WebClient())
+        {
+            foreach (var file in files) 
+            {
+                client.DownloadFile($"https://data.iana.org/time-zones/tzdb-{Version}/{file}", $"{Path.Combine(InputDirectory!, file)}");
+            }
+        }
+
+        files.Remove("zone1970.tab");        
+
+        using (Process process = new Process()) 
+        {
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.FileName = "zic";
+            foreach (var f in files) 
+            {
+                process.StartInfo.Arguments = $"-d \"{OutputDirectory}\" \"{Path.Combine(InputDirectory!, f)}\"";
+                process.Start();
+                process.WaitForExit();
+            }
+        }
+        File.Copy(Path.Combine(InputDirectory!,"zone1970.tab"), Path.Combine(OutputDirectory!,"zone1970.tab"));
+    }
+
+    private void FilterTimeZoneData() 
```

Comment:
Why is this needed? You process additional files and then they are deleted?

Reply:
Yes, this is just to remove extraneous files so that `CreateWasmBundle` doesn't end up including them in the data archive.

Reply:
This method removes unnecessary files so that the subsequent `CreateWasmBundle` does not include them in the data archive.